mlrun/projects/project.py

```diff
@@ -1625,6 +1557,7 @@ def set_function(
             # in hub or db functions name defaults to the function name
             if not name and not (func.startswith("db://") or func.startswith("hub://")):
                 raise ValueError("function name must be specified")
+            image = image or self.default_image
```

Comment:
@theSaarco i don't think thats right, we want to keep the function image None, and set it in the enrichment, so the default_image can still change after we set the functions. especially when we will add the build() option that will be the common case.

Reply:
i guess it will require changes in `_init_function_from_dict()` and to raise an exception in `enrich_function_object()` if there is no image or default_image

Reply:
Yes, I thought about this, but thought this case can be a special case - I modified the code. 
I don't think extra exceptions are needed - `enrich_function_object()` already calls `verify_base_image()` when needed.